agent/structs/config_entry.go

```diff
@@ -479,12 +479,19 @@ func (c *ConfigEntryResponse) MarshalBinary() (data []byte, err error) {
 	bs := make([]byte, 128)
 	enc := codec.NewEncoderBytes(&bs, msgpackHandle)
 
-	if err := enc.Encode(c.Entry.GetKind()); err != nil {
-		return nil, err
-	}
-	if err := enc.Encode(c.Entry); err != nil {
-		return nil, err
+	if c.Entry != nil {
+		if err := enc.Encode(c.Entry.GetKind()); err != nil {
+			return nil, err
+		}
+		if err := enc.Encode(c.Entry); err != nil {
+			return nil, err
+		}
+	} else {
+		if err := enc.Encode(""); err != nil {
```

Comment:
I think this seems ok as long as a real entry cannot have `kind=""`.

Reply:
For singular GET operations it cannot. The only two valid kinds are `proxy-defaults` and `service-defaults`.

For the list endpoint "" is valid to indicate that all entries should be listed. But in the case of marshal/unmarshal here "" is never a valid kind.